safehttp/incoming_request_test.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package safehttp_test
+
+import (
+	"context"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-safeweb/safehttp"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+type pizza struct {
+	val string
+}
+
+type notPizza struct {
+}
+
+type pizzaKey string
+
+func TestRequestSetValidContextWithValue(t *testing.T) {
+	tests := []struct {
+		name    string
+		key     pizzaKey
+		wantVal *pizza
+		wantOk  bool
+	}{
+		{
+			name:    "Value set for key",
+			key:     pizzaKey("1234"),
+			wantOk:  true,
+			wantVal: &pizza{val: "margeritta"},
+		},
+		{
+			name:    "Value not set for key",
+			key:     pizzaKey("5678"),
+			wantOk:  false,
+			wantVal: nil,
+		},
+	}
+	for _, test := range tests {
+		m := safehttp.NewMachinery(func(rw safehttp.ResponseWriter, ir *safehttp.IncomingRequest) safehttp.Result {
+			ctx := context.WithValue(ir.Context(), pizzaKey("1234"), &pizza{val: "margeritta"})
+			ir.SetContext(ctx)
+			got, ok := ir.Context().Value(test.key).(*pizza)
+			if ok != test.wantOk {
+				t.Errorf("type match: got %v, want %v", ok, test.wantOk)
+			}
+			if diff := cmp.Diff(test.wantVal, got, cmp.AllowUnexported(pizza{})); diff != "" {
+				t.Errorf("ir.Context().Value(k): mismatch (-want +got): \n%s", diff)
+			}
+
+			return safehttp.Result{}
+		}, &testDispatcher{})
+
+		rec := newResponseRecorder(&strings.Builder{})
+		req := httptest.NewRequest("GET", "/", nil)
+		m.HandleRequest(rec, req)
+
+		if want := 200; rec.status != want {
+			t.Errorf("response status: got %v, want %v", rec.status, want)
+		}
```

Comment:
Reformat this test so that it doesn't use `Machinery`. You can create an `IncomingRequest` from a `http.Request` using `safehttp.NewIncomingRequest`

Same in the test below